ayanda/src/main/java/sintulabs/p2p/Ayanda.java

```diff
@@ -0,0 +1,94 @@
+package sintulabs.p2p;
+
+import android.content.Context;
+import android.net.wifi.p2p.WifiP2pDevice;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Created by sabzo on 1/19/18.
+ */
+
+public class Ayanda {
+    private Bluetooth bt;
+    private Lan lan;
+    private WifiDirect wd;
+
+    private Context context;
+
+    public Ayanda(Context context, IBluetooth iBluetooth, ILan iLan, IWifiDirect iWifiDirect) {
+        this.context = context;
+        bt = new Bluetooth(context, iBluetooth);
+        lan = new Lan(context, iLan);
+        wd = new WifiDirect(context, iWifiDirect);
+    }
+
+    /*
+        Discover nearby devices that have made themselves detectable blue Bluetooth
+        Discovers nearby devices and stores them in a collection of devices found.
+     */
+    public void btDiscover() {
```

Comment:
Related to my comment above --- don't expose any protocol-specific methods here. This can all be hidden from the user. Instead have single methods like `enable()`, `disable()`, `discover()`, etc.